### ISSUE TITLE
xmpp streams disconnects when namespaces are not well formed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4327]
|<img alt="sciasbat's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| sciasbat reported|
|-|-|
|Trac ID|trac#4327|
|Type|defect|
|Created|2010-02-27 20:43:59Z|

Though the behavior is in theory correct, due to other buggy software the xml stream disconnects when receiving tags with trailing spaces in namespaces uris. Example:

<tag xmlns="uri:sometring "/>

A trivial fix has been proposed by Ralph: 

http://www.mail-archive.com/twisted-jabber@ik.nu/msg00264.html

<details><summary>Searchable metadata</summary>

```
trac-id__4327 4327
type__defect defect
reporter__sciasbat sciasbat
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__words words
keywords__ 
time__1267303439000000 1267303439000000
changetime__1297843784000000 1297843784000000
version__None None
owner__ 
cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @ralphm|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="awclin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|awclin set owner to awclin<br>awclin set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="awclin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|awclin removed owner<br>awclin set status to `new`|
|-|-|

The fix for [#4503](https://github.com/twisted/twisted/issues/4503) applies (amongst other things) the exact fix outlined in the mailing list thread referenced by the reporter. We have ```test_namespaceWithWhitespace``` in words.test.test_domish also, which seems to provide adequate cover for this issue.  

Even if this ticket remains open due to the larger (potential) issue around whether we should actively drop support for namespace-well-formedness (some discussion of this is in the linked mailing list thread), I think this one can be closed, and we can open a more relevant one if needs be.

P.S. Not sure on procedure for this kind of resolution to a ticket, so putting it up for review

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm set status to `closed`|
|-|-|

Agreed. While the summary of this ticket is slightly misleading (as explained in [[#4503](https://github.com/twisted/twisted/issues/4503)](https://github.com/twisted/twisted/issues/4503), the value may have spaces), this ticket can be closed as having been resolved by r29852 and being a (reverse) duplicate of #4503.

The discussion on _real_ namespace-well-formedness could eventually spawn a new ticket.
